runtime/queries/metricsview_aggregation.go

```diff
@@ -417,7 +417,12 @@ func (q *MetricsViewAggregation) buildMetricsAggregationSQL(mv *runtimev1.Metric
 
 	groupCols := make([]string, 0, len(q.Dimensions))
 	unnestClauses := make([]string, 0)
+	joinConditions := make([]string, 0, len(q.Dimensions))
+	selfJoinCols := make([]string, 0, len(q.Dimensions)+1)
+	selfJoinTableAlias := tempName("self_join")
+	nonNullValue := tempName("non_null")
 	var args []any
+	var subSelectArgs []any
```

Comment:
I'm afraid this takes the code to a barely readable level and will make future changes too hard and error-prone. I don't have a great proposal, but somehow this filter case needs to be isolated better in the code (and not interleaved with the existing implementation with various if statements throughout) and also have an appropriate docstring explaining the case being handled.

I wonder if it can either be refactored to a single `if` statement at the end of the `buildMetricsAggregationSQL` function, or otherwise we may need to have a dedicated `buildMetricsAggregationSingleMeasureFilterSQL` function that gets invoked instead?

Reply:
`args` collection prevents from refactoring it to a single `if`